src/domain/participants/participants.js

```diff
@@ -170,49 +182,61 @@ const postParticipantsBatch = async (req) => {
+            value: party.partyIdentifier
+          }]).toApiErrorObject())
         }
       }
-      for (let [key, value] of typeMap) {
+      for (const [key, value] of typeMap) {
         const payload = {
           requestId: requestId,
           partyList: value
         }
         Logger.info(`postParticipantsBatch::oracleBatchRequest::type=${key}`)
-        const response = await oracle.oracleBatchRequest(req, key, payload)
+        const response = await oracle.oracleBatchRequest(headers, method, requestPayload, key, payload)
         if (response && (response.data !== null || response.data !== undefined)) {
           if (Array.isArray(response.data.partyList) && response.data.partyList.length > 0) {
             for (const party of response.data.partyList) {
               party.partyId.currency = undefined
               overallReturnList.push(party)
             }
           } else {
-            for (let party of value) {
-              overallReturnList.push(util.buildBatchErrorObject(party, Errors.ErrorObject.ADD_PARTY_ERROR, [{key: party.partyIdType, value: party.partyIdentifier}]))
+            for (const party of value) {
+              overallReturnList.push(ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, undefined, undefined, undefined, [{
+                key: party.partyIdType,
+                value: party.partyIdentifier
+              }]).toApiErrorObject())
             }
           }
         } else {
-          // TODO: what happens when nothing is returned
           for (const party of value) {
-            overallReturnList.push(util.buildBatchErrorObject(party, Errors.ErrorObject.ADD_PARTY_ERROR, [{key: party.partyIdType, value: party.partyIdentifier}]))
+            overallReturnList.push(ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ADD_PARTY_INFO_ERROR, undefined, undefined, undefined, [{
+              key: party.partyIdType,
+              value: party.partyIdentifier
+            }]).toApiErrorObject())
           }
         }
       }
       const payload = {
         partyList: overallReturnList,
-        currency: req.payload.currency
+        currency: requestPayload.currency
       }
-      await participant.sendRequest(req, req.headers['fspiop-source'], Enums.endpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT, Enums.restMethods.PUT, payload, {requestId})
+      await participant.sendRequest(headers, headers[Enums.Http.Headers.FSPIOP.SOURCE], Enums.EndPoints.FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTICIPANT_BATCH_PUT, Enums.Http.RestMethods.PUT, payload, {requestId})
       Logger.info('postParticipantsBatch::end')
     } else {
       Logger.error('Requester FSP not found')
-      // TODO: handle issue where requester fsp not found send to error handling framework
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()
```

Comment:
`throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found').toApiErrorObject()` --> `throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found')`